### ISSUE TITLE
Allow omitting semicolons before braces

### ISSUE THREAD

**Issue by hgl** [eslint/eslint#2622]
This should be pretty common with arrow functions:
```js
let a = () => {if (foo) bar()};
```

Currently enable `semi` requires a semicolon after `bar()`, and since space is usually required after semicolon, you need to write it like
```js
let a = () => {if (foo) bar(); };
```

This doesn't look balanced.

Do you think we should allow omitting semicolons here with `semi` enabled?

**Comment by michaelficarra** [comment]
You wouldn't write `function a() {if (foo) bar()}`, would you? Why should function bodies in arrows be treated differently?

**Comment by hgl** [comment]
I probably would for function expressions:

```js
foo(function (a) {if (bar) baz(a)}); 
```

**Comment by michaelficarra** [comment]
Then you should turn off the rule because it is not for you.

**Comment by ilyavolodin** [comment]
Agree with @michaelficarra While this might be stylistic preference, this rule is there to make sure you put semicolons everywhere where they should exist. In this case, they are required (even though omitting them will not chance anything).
